Bluetooth: Controller: Use macros for access address and CRC init sizes

ssam18 opened the pull request:
## Description

This PR addresses issue #40378 by replacing hardcoded array size literals with symbolic constants for  and  arrays in the Bluetooth controller code.

## Changes

1. **New define added** ():
   -  (3 bytes) - Specifically for CRC initialization values

2. **Updated files**:
   -  - 6 instances
   -  - 6 instances

cvinayak reviewed (commented):
LGTM. Nitpick, reorder variable definitions in the inverse order of their line length (reverse christmas tree).

cvinayak reviewed (changes requested):
Seems you have randomly shuffled the lines.  See how LHS has the declarations ordered by line length before your change.

cvinayak reviewed (dismissed):
Comments need to be addressed.

ssam18 committed:
Bluetooth: Controller: Use macros for access address and CRC init sizes. Replace hardcoded array size literals with symbolic constants for access_addr and crc_init arrays in the Bluetooth controller code.

Added PDU_CRCINIT_SIZE (3 bytes) in pdu.h for CRC initialization values. Updated lll_adv_iso.c and lll_sync_iso.c to use PDU_ACCESS_ADDR_SIZE and PDU_CRCINIT_SIZE in local variable declarations. Reordered variable declarations to follow reverse christmas tree style.

Signed-off-by: Samaresh Kumar Singh <ssam3003@gmail.com>
